### PR TITLE
Some tweaks I made to the code I hope it help!

### PR DESCRIPTION
**Pull request by triorr**
* Reorderable columns
* Library menu item "Add and play"
* And some other tweaks

### CODE REVIEW

**Comment by matiasdelellis** [comment]
Hi @triorr 
Thank you very much for the contribution! :wink: 
I make a couple of comments within the code that you should improve..

File: src/pragha-playlist.c [hunk 1]
```diff
@@ -2246,6 +2247,49 @@ pragha_playlist_insert_mobj_list(PraghaPlaylist *cplaylist,
...
+	gtk_tree_view_set_model(GTK_TREE_VIEW(cplaylist->view), NULL);
+
+	gtk_tree_view_set_model(GTK_TREE_VIEW(cplaylist->view), cplaylist->model);
+
+	gint i=0;
+	for (l = list; l != NULL; l = l->next) {
+		mobj = l->data;
+		if(i==0){
+			pragha_playlist_append_mobj_and_play(cplaylist, mobj);
+
```
Comment (matiasdelellis) [inline src/pragha-playlist.c]: About code:
 * Should save the reference to the first mobj, add the entire list, and play the first mobj.

About styles:
 * Space between assignments and comparisons..  e.g: 'if(i==0){' -> 'if (i == 0){'  and 'gint i=0;'   ->  'gint i = 0;'
 * Do not leave unnecessary blank lines ..
 * spaces between brackets and conditionals..  e.g: '}else{'     ->     '} else {'

File: src/pragha-playlist.c [hunk 2]
```diff
@@ -2246,6 +2247,49 @@ pragha_playlist_insert_mobj_list(PraghaPlaylist *cplaylist,
...
+	
+
+	pragha_playlist_set_changing(cplaylist, FALSE);
+	remove_watch_cursor (GTK_WIDGET(cplaylist));
+
+	g_signal_emit (cplaylist, signals[PLAYLIST_CHANGED], 0);
+
+	if(gtk_tree_sortable_get_sort_column_id(GTK_TREE_SORTABLE(cplaylist->model),
+	                                        &column, &order))
+		select_numered_path_of_current_playlist(cplaylist, prev_tracks, TRUE);
```
Comment (matiasdelellis) [inline src/pragha-playlist.c]: I do not understand why you need to do this .. :disappointed: 
Please explain it.. :wink:

File: src/pragha-playlist.c [hunk 3]
```diff
@@ -3611,237 +3632,54 @@ create_current_playlist_columns(PraghaPlaylist *cplaylist, GtkTreeView *view)
...
+	column = gtk_tree_view_column_new_with_attributes(label,
 							  renderer,
 							  "text",
-							  P_FILENAME,
+							  column_id,
 							  NULL);
 	gtk_tree_view_column_set_sizing(column, GTK_TREE_VIEW_COLUMN_FIXED);
 	gtk_tree_view_column_set_resizable(column, TRUE);
-	gtk_tree_view_column_set_sort_column_id(column, P_FILENAME);
+	gtk_tree_view_column_set_reorderable(column, TRUE);
```
Comment (matiasdelellis) [inline src/pragha-playlist.c]: Is interesting to be able to sort the columns .. But I should be able to keep the order between restarts of pragha ... :wink: 
Do you dare to do it? :stuck_out_tongue_winking_eye:

File: src/pragha-playlist.c (hunk 2 above)
Comment (triorr) [reply src/pragha-playlist.c]: Too much copy&paste :laughing:

File: src/pragha-playlist.c (hunk 3 above)
Comment (triorr) [reply src/pragha-playlist.c]: I will try later . It breaks the add and delete columns to.

File: src/pragha-playlist.c (hunk 1 above)
Comment (triorr) [reply src/pragha-playlist.c]: Thanks for your reply . 
Check  this commit [83b636f ](https://github.com/pragha-music-player/pragha/pull/131/commits/83b636f01cd8da8f6c193aa736c4aa508e2b60dd) please.

**Comment by matiasdelellis** [comment]
Well..
I will wait if you can complete the reordering of columns .. otherwise we separate it in another pull request.. 
Thanks again.. :wink:

**Comment by matiasdelellis** [comment]
Wow ..
Except for a bunch of style complaints -That I'll let it pass- i love the result .. :smile: 

Just try it and it works ... Something pending?

**Comment by triorr** [comment]
Nice , I,m glade you liked the result .
Style?
No nothing pending . 
I can make a pull request if you want.

**Comment by matiasdelellis** [comment]
MMmm.. Coding style .. :wink:

Congrats.. your first contribute to Pragha.. :smile: 

Thanks again.. 
Regards,
Matias.

**Comment by triorr** [comment]
My first contribute ever !!
:smile:

**Pull request merged**